services/user-service/utils/passport.ts

```diff
@@ -0,0 +1,127 @@
+import passport from 'passport';
+import LocalStrategy from 'passport-local';
+import FacebookStrategy from "passport-facebook";
+import { UserAuthTypeEnum } from '../utils/enum';
+import User from '../models/user-model';
+import UserAuth from '../models/user-auth-model';
+import sequelize from './sequelize';
+import { Transaction } from 'sequelize';
+
+const strategy = {
+  login: async (profileId: string, authType: UserAuthTypeEnum, options: {
+    user: {
+      firstName: string,
+      lastName?: string | null | undefined,
+      email: string,
+      imageUrl?: string | null | undefined,
+      auth: {
+        profileId: string,
+        accessToken?: string | undefined, 
+        refreshToken?: string| undefined,
+        data?: object | undefined,
+      }
+    }
+  }) => {
+    const userAuth = await UserAuth.findOne({
+      where: {
+        authType: authType,
+        username: profileId,
+      },
+    });
+
+    const userToViewModel = (user: User) => {
+      return {
+        id: user.id,
+        firstName: user.firstName,
+        lastName: user.lastName,
+      }
+    }
+
+    if(userAuth) {
+      const user = await User.findOne({
+        where: {
+          id: userAuth.userId,
+        }
+      });
+
+      if(user == null) {
+        throw new Error('User not found');
+      };
+
+      return userToViewModel(user);
+    } else {
+      await sequelize.transaction(async (t: Transaction) => {
+        const user = await User.create({
+          firstName: options.user.firstName,
+          lastName: options.user.lastName,
+        }, {transaction: t});
+  
+        await UserAuth.create({
+          userId: user.id || 0,
+          authType: authType,
+          username: options.user.auth.profileId,
+          profileData: options.user.auth.data ? JSON.stringify(options.user.auth.data) : undefined,
+        }, {transaction: t});
+
+        return userToViewModel(user);
+      });
+    }
+  },
+
+  localStrategy: new LocalStrategy.Strategy(
+    {
+      usernameField: 'email',
+      passwordField: 'password',
+    }, 
+    async(email: string, password: string, done) => {
+
+    }
+  ),
+
+  facebookStrategy: new FacebookStrategy.Strategy({
+    clientID: process.env.USER_SERVICE_FACEBOOK_CLIENT_ID as string,
+    clientSecret: process.env.USER_SERVICE_FACEBOOK_CLIENT_SECRET as string,
```

Comment:
we need to prevent re type casting on the env file in here, will bring the follow up later regarding this issues